src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -380,24 +391,25 @@ uct_ib_mlx5_ep_set_rdma_seg(struct mlx5_wqe_raddr_seg *raddr, uint64_t rdma_radd
 
 static UCS_F_ALWAYS_INLINE void
 uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg, int is_global,
-                          struct mlx5_wqe_av *av, uint8_t path_bits)
+                          uct_ib_mlx5_base_av_t *av,
+                          struct mlx5_grh_av *grh_av, uint8_t path_bits)
```

Comment:
no - when passing NULL it could be determined immediately. When passing pointer there would still be 1 runtime branch.